src/main/java/ro/unibuc/hello/controller/TaskController.java

```diff
@@ -0,0 +1,73 @@
+package ro.unibuc.hello.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.BeanUtils;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ro.unibuc.hello.data.*;
+import ro.unibuc.hello.dto.CreateTask;
+import ro.unibuc.hello.dto.TaskDto;
+import ro.unibuc.hello.exception.BadRequestException;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@RequestMapping("/task")
+@RequiredArgsConstructor
+@RestController
+public class TaskController {
+
+	final TaskRepository taskRepository;
+	final ObjectMapper objectMapper;
+
+	@GetMapping("/{id}")
+	public TaskDto getById (@PathVariable String id){
+		Optional<Task> optionalTask = taskRepository.findById(id);
+		if(optionalTask.isPresent()){
+			TaskDto taskDto = new TaskDto();
+			BeanUtils.copyProperties(optionalTask.get(), taskDto);
+			return taskDto;
+		} else {
+			throw new BadRequestException("There is no task with this id!");
+		}
+	}
+
+	@GetMapping("/all")
```

Comment:
I think we should stick to RESTful API naming, so instead of "/all" using just "/" would fit.